src/index/CompressedRelation.cpp

```diff
@@ -790,8 +791,8 @@ CPP_template_def(typename IdGetter)(
   // Iterate over the blocks and only read (and decompress) those which
   // contain more than one different `colId`. For the others, we can determine
   // the count from the metadata.
-  for (size_t i = 0; i < relevantBlocksMetadata.size(); ++i) {
-    const auto& blockMetadata = relevantBlocksMetadata[i];
+  size_t i = 0;
+  for (const auto& blockMetadata : ql::views::join(relevantBlockMetadata)) {
```

Comment:
You can use `for (const auto& [i, blockMetadata] : ::ranges::views::enumerate(relevantBla)`.
(from `range-v3`.